### DISCUSSION TITLE
Water Flow Sensor Count issue

### DISCUSSION THREAD

**Discussion by Dreamdancer100** [arendst/Tasmota#21925] [Q&A]
Hello, I have a problem with my counter2 ! I have a solenoid valve that is controlled by a float switch and there is a flow meter in the water pipe. This is supposed to count the water using a tasmota counter. which he had been doing until recently. The whole thing is controlled via iobroker and a blockly script and the counter is evaluated and calibrated. 
How can I reset the counter so that the pulse time runs normally again? I had a pulse value of 192 for one liter, which I had calibrated to milliliters, but now when the float drops and the solenoid valve switches on, the counter no longer counts the water. what can I do.
I’ve found on internet these code and think that is the problem I type these into console but after these the counter count only 1 + 1 
`switchmode1 1
counter2 0
rule1 on switch1#state=0 do Counter2 +1 endon       
rule1 5
rule1 1 `
The whole thing is controlled via iobroker and a blockly script and the counter is evaluated and calibrated
I actually wanted to limit the float to 60 seconds so that too much water doesn't run into the tank. But that's complete nonsense, because the negative line from the solenoid valve runs over the float, so the float can do what it's supposed to do without a time limit. It should only be false when the device starts and the flow meter should measure its count normally via counter

**Comment by lalo-uy** [comment]
I don’t get the problem.
You want to reset the counter every 196 counts? With a rule:
ON counter#c1>196 DO …….

**Reply by sfromis** [reply]
Using `SetOption79`, Tasmota can reset the counter every telemetry period, after reporting the data for the period.

**Reply by Dreamdancer100** [reply]
I've tried a lot of things and SetOption79 unfortunately didn't work. The counter still counts 1+1 or existing value +1

**Reply by sfromis** [reply]
"Didn't work" is to keep secret what you are really wanting to do. Anyway, when I mentioned to enable `SetOption79` (not just look at the value) it was not meant as "this will get you what you want" (whatever that really is), but only to add a piece of info about how you can get the counter reset **at the telemetry period**, which by default is every 5 minutes, after the value was published in the regular payload at the interval.

**Reply by Dreamdancer100** [reply]
Would it help to completely re-flash and set up the esp!?

**Reply by sfromis** [reply]
That depend on what you've been doing. In some cases, users may have been trying too many things at the same time, where a fresh start may be useful, like doing a full reset (doing flashing again makes no difference outside of this implicitly also doing a reset). But as you've hardly posted details, or done debugging, you can't get clear answers about what to do for what you define as what you want.

**Reply by Dreamdancer100** [reply]
when i activate sensor debug get this 
`09:23:26.075 MQT: stat/Aqua_Float/RESULT = {"Counter1":-99}
09:23:30.239 MQT: tele/Aqua_Float/STATE = {"Time":"2024-08-18T09:23:30","Uptime":"0T00:07:09","UptimeSec":429,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"GordonX-Fritzbox","BSSId":"2C:91:AB:57:6F:7A","Channel":11,"Mode":"11n","RSSI":70,"Signal":-65,"LinkCount":1,"Downtime":"0T00:00:04"}}
09:23:30.267 MQT: tele/Aqua_Float/SENSOR = {"Time":"2024-08-18T09:23:30","Switch1":"ON","COUNTER":{"C1":4294967197},"SR04":{"Distance":22.1}}
09:23:45.252 MQT: tele/Aqua_Float/STATE = {"Time":"2024-08-18T09:23:45","Uptime":"0T00:07:24","UptimeSec":444,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"GordonX-Fritzbox","BSSId":"2C:91:AB:57:6F:7A","Channel":11,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:04"}}
09:23:45.282 MQT: tele/Aqua_Float/SENSOR = {"Time":"2024-08-18T09:23:45","Switch1":"ON","COUNTER":{"C1":4294967197},"SR04":{"Distance":22.1}}
09:24:00.217 MQT: tele/Aqua_Float/STATE = {"Time":"2024-08-18T09:24:00","Uptime":"0T00:07:39","UptimeSec":459,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"GordonX-Fritzbox","BSSId":"2C:91:AB:57:6F:7A","Channel":11,"Mode":"11n","RSSI":70,"Signal":-65,"LinkCount":1,"Downtime":"0T00:00:04"}}
09:24:00.245 MQT: tele/Aqua_Float/SENSOR = {"Time":"2024-08-18T09:24:00","Switch1":"ON","COUNTER":{"C1":4294967197},"SR04":{"Distance":22.0}}
09:24:15.240 MQT: tele/Aqua_Float/STATE = {"Time":"2024-08-18T09:24:15","Uptime":"0T00:07:54","UptimeSec":474,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"GordonX-Fritzbox","BSSId":"2C:91:AB:57:6F:7A","Channel":11,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:04"}}
09:24:15.270 MQT: tele/Aqua_Float/SENSOR = {"Time":"2024-08-18T09:24:15","Switch1":"ON","COUNTER":{"C1":4294967197},"SR04":{"Distance":22.4}}
09:24:30.217 MQT: tele/Aqua_Float/STATE = {"Time":"2024-08-18T09:24:30","Uptime":"0T00:08:09","UptimeSec":489,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"GordonX-Fritzbox","BSSId":"2C:91:AB:57:6F:7A","Channel":11,"Mode":"11n","RSSI":74,"Signal":-63,"LinkCount":1,"Downtime":"0T00:00:04"}}
09:24:30.245 MQT: tele/Aqua_Float/SENSOR = {"Time":"2024-08-18T09:24:30","Switch1":"ON","COUNTER":{"C1":4294967197},"SR04":{"Distance":21.5}}`
I don't know what you have to enter into the console to get the counter value or other settings

**Reply by Dreamdancer100** [reply]
I re-flashed the ESP once but it stays the same - only 1+1 is counted per pumping process. Can't you reset it to standard so that a value like 2200 comes out again with about 1 liter pumped!?

**Reply by Dreamdancer100** [reply]
So I found the problem. I had set the switch and counter settings backwards. Now it is set correctly, only the running time for the switch is limited. This means that if the float is true, the solenoid valve should also be set to true and the counter counts the pulse. only the true time is limited by the switch. can you change the true time!?
